solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -3178,6 +3179,9 @@ public <T> T initPlugins(
             type.getSimpleName() + "." + info.name, (SolrMetricProducer) o);
       }
       if (o instanceof CircuitBreaker) {
+        if (o instanceof SolrCoreAware) {
+          ((SolrCoreAware) o).inform(this);
```

Comment:
As this init mechanism requires a default no-arg constructor, had to make it SolrCoreAware to inject CC.